# Table of contents

* [Bioinformatics Tutorial - Basic](README.md)
* [Getting Started](getting-started.md)

## Part I. Basic Skills

* [1.Setup](part-i.-basic-skills/1.setup/README.md)
  * [1.1 Docker](part-i.-basic-skills/1.setup/1.1.docker.md)
  * [1.2 Cluster](part-i.-basic-skills/1.setup/1.2-cluster.md)
* [2.Linux](part-i.-basic-skills/2.linux/README.md)
  * [2.1 Basic Command](part-i.-basic-skills/2.linux/2.1.linux-basic-command.md)
  * [2.2 Practice Guide](part-i.-basic-skills/2.linux/2.2.linux-practice-guide.md)
* [3.R](part-i.-basic-skills/3.r/README.md)
  * [3.1 R Basics](part-i.-basic-skills/3.r/3.1.r-basics.md)
  * [3.2 Plot with R](part-i.-basic-skills/3.r/3.2.plots-with-r.md)
* [4.Python](part-i.-basic-skills/4.python.md)

## PART II. BASIC ANALYSES

* [1.Blast](part-ii.-basic-analyses/1.seqblast.md)
* [2.Conservation Analysis](part-ii.-basic-analyses/2.conservation-analysis.md)
* [3.Function Analysis](part-ii.-basic-analyses/3.function-analysis/README.md)
  * [3.1 GO](part-ii.-basic-analyses/3.function-analysis/3.1.go.md)
  * [3.2 KEGG](part-ii.-basic-analyses/3.function-analysis/3.2.kegg.md)
  * [3.3 GSEA](part-ii.-basic-analyses/3.function-analysis/3.3.gsea.md)

## Part III. NGS DATA ANALYSES

* [1.Mapping](part-iii.-ngs-data-analyses/1.mapping.md)
* [2.RNA-seq](part-iii.-ngs-data-analyses/2.rna-seq/README.md)
  * [2.1 Differential Expression](part-iii.-ngs-data-analyses/2.rna-seq/2.1.differential-expression.md)
  * [2.2 Alternative Splicing](part-iii.-ngs-data-analyses/2.rna-seq/2.2.alternative-splicing.md)
* [3.ChIP-seq](part-iii.-ngs-data-analyses/3.chip-seq.md)
* [More Analyses](part-iii.-ngs-data-analyses/4.more-analyses.md)
* [4.Network](part-iii.-ngs-data-analyses/4.network/README.md)
  * [4.1.Co-expression Network](https://yuminthu.github.io/training_class/docs/co_expression.html)
  * [4.2.miRNA Targets](https://yuminthu.github.io/training_class/docs/miRNA.html)
  * [4.3.RBP-RNA Interactions](https://yuminthu.github.io/training_class/docs/RBP_interaction.html)
* [5.Motif](part-iii.-ngs-data-analyses/5.motif/README.md)
  * [5.1.Sequence Motif](https://yuminthu.github.io/training_class/docs/sequence_motif.html)
  * [5.2.Structure Motif](https://yuminthu.github.io/training_class/docs/structure_motif.html)
* [6.RNA Regulation Analyses](part-iii.-ngs-data-analyses/6.rna-regulation-analyses/README.md)
  * [6.1.RNA Editing](https://yuminthu.github.io/training_class/docs/RNA_editing.html)
  * [6.2.APA \(Alternative Polyadenylation\)](https://yuminthu.github.io/training_class/docs/APA.html)
  * [6.3.Ribo-seq](https://yuminthu.github.io/training_class/docs/ribo_seq.html)
  * [6.4.Structure-seq](https://yuminthu.github.io/training_class/docs/structure_seq.html)
  * [6.5.Modification \(m6A-seq\)](part-iii.-ngs-data-analyses/6.rna-regulation-analyses/6.5.modification-m6a-seq.md)
* [7.Clinical Analyses](part-iii.-ngs-data-analyses/7.clinical-analyses/README.md)
  * [7.1.ROC Curve](https://yuminthu.github.io/training_class/docs/ROC_curve.html)
  * [7.2.PCA/tSNE](part-iii.-ngs-data-analyses/7.clinical-analyses/PCA_tSNE.md)
  * [7.3.Survival Analysis](https://yuminthu.github.io/training_class/docs/survival_analysis.html)

## Part IV. MACHINE LEARNING

* [1.Machine Learning Basics](part-iv.-machine-learning/1.machine-learning-basics.md)
* [2. Machine Learning with R](part-iv.-machine-learning/2.machine-learning-with-r.md)
* [3. Machine Learning with Python](part-iv.-machine-learning/3.machine-learning-with-python.md)

## Part V. Quiz

* [1. exRNA Biomarker](part-v.-quiz/1.quiz_exrna/README.md)
  * [1.1 Data Introduction](part-v.-quiz/1.quiz_exrna/1.1-data-introduction.md)
  * [1.2 Requirement](part-v.-quiz/1.quiz_exrna/1.2-requirement.md)
  * [1.3 Helps](part-v.-quiz/1.quiz_exrna/1.3-helps.md)

## Appendix

* [Appendix I. Keep Learning](appendix/appendix1.keep-learning.md)
* [Appendix II. Databases & Servers](appendix/appendix2.databases-servers.md)

