# 2.RNA-seq

* [**Differential expression**](https://lulab2.gitbook.io/teaching/part-iii.-ngs-data-analyses/2.rna-seq/2.1.differential-expression)
* [**Alternative splicing**](https://lulab2.gitbook.io/teaching/part-iii.-ngs-data-analyses/2.rna-seq/2.2.alternative-splicing)

## More Reading: RNA-seq Analysis Pipelines

* Trapnell C et al. Nat Protoc. 2012, 7\(3\):562-78 Differential gene and transcript expression analysis of RNA-seq experiments with TopHat and Cufflinks. [Reference Link](http://www.ncbi.nlm.nih.gov/pubmed/22383036)
* [A survey of best practices for RNA-seq data analysis -- 2016](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8#Abs1)
* RNA-seq analysis pipeline at GitHub: [https://github.com/mgonzalezporta/TeachingMaterial](https://github.com/mgonzalezporta/TeachingMaterial)
* lncRNA analysis pipeline: [http://webhome.weizmann.ac.il/home/igoru/PLAR/](http://webhome.weizmann.ac.il/home/igoru/PLAR/)
* \(Deprecated\) RNA-seq analysis pipeline @ Lu Lab: [RNAfinder](http://bioinformatics.life.tsinghua.edu.cn/new_home/lulab-software/rnafinder.html)
